PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,3 +1,11 @@
+## Checklist 🛎️
+
+- [ ] Add product, design, and eng for rdev review
```

Comment:
Do this mean as PR reviewers?

Reply:
Yes! I updated the sentence to: `Add product, design, and eng as reviewers for rdev review` does that look better?